lib/datadog/tracing/contrib/aws/instrumentation.rb

```diff
@@ -60,6 +64,96 @@ def annotate!(span, context)
             span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_STATUS_CODE, context.safely(:status_code))
           end
 
+          def add_service_specific_tags(span, aws_service, params)
+            case aws_service
+            when 'sqs'
+              add_sqs_tags(span, params)
+            when 'sns'
+              add_sns_tags(span, params)
+            when 'dynamodb'
+              add_dynamodb_tags(span, params)
+            when 'kinesis'
+              add_kinesis_tags(span, params)
+            when 'eventbridge'
+              add_eventbridge_tags(span, params)
+            when 'states'
+              add_states_tags(span, params)
+            when 's3'
+              add_s3_tags(span, params)
+            end
+          end
+
+          def add_sqs_tags(span, params)
+            queue_url = params.fetch(:queue_url, nil)
+            queue_name = params.fetch(:queue_name, nil)
+            if queue_url
+              # example queue_url: https://sqs.sa-east-1.amazonaws.com/12345678/MyQueueName
+              queue_name = queue_url.split('/')[-1]
+              aws_account = queue_url.split('/')[-2]
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_QUEUE_NAME, queue_name)
```

Comment:
String operations are expensive.
We are splitting `queue_url` twice here. We should only split it once then collect the result.

Also, I recommend being more strict with this splitting, given invalid URLs are unlikely to happen here. And if they do, we probably want to skip tagging.

```suggestion
            if queue_url
              _, _, _, aws_account, queue_name = queue_url.split('/')
              if aws_account && queue_name
                span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
                span.set_tag(Ext::TAG_QUEUE_NAME, queue_name)
              end
            end
```

Reply:
I tried adding your suggested change but it would incorrectly add the queue_name and aws_account when given an invalid queue_url. I wrote a unit test for this case. I ended up just doing a regex match